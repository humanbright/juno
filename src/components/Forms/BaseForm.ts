import { FilledInputProps } from '@material-ui/core/FilledInput';
import { InputProps } from '@material-ui/core/Input';
import { InputLabelProps } from '@material-ui/core/InputLabel';
import { OutlinedInputProps } from '@material-ui/core/OutlinedInput';

type RcBaseFormProps = {
  InputProps?:
    | Partial<InputProps>
    | Partial<FilledInputProps>
    | Partial<OutlinedInputProps>;
  inputProps?: InputProps['inputProps'];
  InputLabelProps?: Partial<InputLabelProps>;

  fullWidth?: boolean;
  readonly?: boolean;
};

export { RcBaseFormProps };
