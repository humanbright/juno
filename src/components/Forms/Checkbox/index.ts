export * from './Checkbox';
export * from './styles';
export * from './utils';
