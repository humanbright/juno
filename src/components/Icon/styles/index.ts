export * from './IconStyle';
export * from './iconSvg';
