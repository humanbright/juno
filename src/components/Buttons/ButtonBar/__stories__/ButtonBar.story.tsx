import { boolean, number, select } from '@storybook/addon-knobs';
import React from 'react';

import refresh from '../../../../icon/Refresh';
import star from '../../../../icon/Star';
import starBorder from '../../../../icon/StarBorder';
import { RcCheckbox } from '../../../Forms/Checkbox';
import { RcIcon } from '../../../Icon';
import { RcIconButton } from '../../IconButton/IconButton';
import { RcButtonBar, RcButtonBarProps } from '../ButtonBar';

export default {
  title: 'ðŸ–¤ deprecated Components/Buttons/ButtonBar',
};

function getKnobs() {
  const direction = select<RcButtonBarProps['direction']>(
    'direction',
    {
      vertical: 'vertical',
      horizontal: 'horizontal',
    },
    'horizontal',
  );

  const size = select(
    'size',
    {
      small: 'small',
      medium: 'medium',
      large: 'large',
    },
    'medium',
  );

  const overlapSize = number('overlapSize', 0);
  const invisible = boolean('invisible', false);
  return {
    direction,
    size,
    overlapSize,
    invisible,
  };
}

export const PlainIconButtonBar = () => {
  return (
    <div>
      <RcButtonBar {...getKnobs()}>
        <RcIconButton variant="plain" title="like" symbol={refresh} />
        <RcIconButton variant="plain" title="like" symbol={refresh} />
        <RcIconButton variant="plain" title="like" symbol={refresh} />
        <RcIconButton variant="plain" title="like" symbol={refresh} />
      </RcButtonBar>
    </div>
  );
};

export const RoundIconButtonBar = () => {
  return (
    <div>
      <RcButtonBar {...getKnobs()}>
        <RcIconButton variant="round" title="like" symbol={refresh} />
        <RcIconButton variant="round" title="like" symbol={refresh} />
        <RcIconButton variant="round" title="like" symbol={refresh} />
        <RcIconButton variant="round" title="like" symbol={refresh} />
      </RcButtonBar>
    </div>
  );
};

export const CheckboxButtonBar = () => {
  return (
    <div>
      <RcButtonBar {...getKnobs()}>
        <RcCheckbox
          title="favorite"
          useRcTooltip
          icon={<RcIcon symbol={starBorder} />}
          checkedIcon={<RcIcon symbol={star} />}
        />
        <RcCheckbox
          title="favorite"
          useRcTooltip
          icon={<RcIcon symbol={starBorder} />}
          checkedIcon={<RcIcon symbol={star} />}
        />
        <RcCheckbox
          title="favorite"
          icon={<RcIcon symbol={starBorder} />}
          checkedIcon={<RcIcon symbol={star} />}
        />
        <RcCheckbox
          title="favorite"
          icon={<RcIcon symbol={starBorder} />}
          checkedIcon={<RcIcon symbol={star} />}
        />
      </RcButtonBar>
    </div>
  );
};
