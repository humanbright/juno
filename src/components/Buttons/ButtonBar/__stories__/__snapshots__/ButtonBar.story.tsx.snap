// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸ–¤ deprecated Components/Buttons/ButtonBar Checkbox Button Bar 1`] = `
.c3 {
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  pointer-events: none;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  white-space: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0px;
}

.c0.c0 > *:nth-child(n + 2) {
  margin-left: 0;
}

.c1.RcCheckbox-root {
  color: #939393;
  padding: 10px;
}

.c1.RcCheckbox-root.RcCheckbox-checked {
  color: #066FAC;
}

.c1.RcCheckbox-root.RcCheckbox-disabled {
  color: #B9B9B9;
}

@media (hover:hover) and (pointer:fine) {
  .c1.RcCheckbox-root:hover {
    background-color: rgba(147,147,147,0.12);
  }

  .c1.RcCheckbox-root:hover.RcCheckbox-checked {
    color: rgba(6,111,172,0.92);
    background-color: rgba(6,111,172,0.12);
  }
}

<div>
  <div
    className="c0 "
    direction="horizontal"
    onClick={[Function]}
    onKeyUp={[Function]}
    size="medium"
  >
    <span
      aria-describedby={null}
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiCheckbox-root RcCheckbox-root c1"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
      title="favorite"
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          className="PrivateSwitchBase-input-18"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c2 star_border icon"
        >
          <svg
            className="c3"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 00.753.547l6.641.965a2.001 2.001 0 011.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 01-2.902 2.108l-5.94-3.123a1.002 1.002 0 00-.931 0l-5.94 3.123a2 2 0 01-2.902-2.109l1.134-6.614a.999.999 0 00-.288-.885l-4.806-4.684a2 2 0 011.109-3.411l6.641-.965a1 1 0 00.753-.547l2.97-6.018a2 2 0 012.679-.908zM13.03 9.537a3.001 3.001 0 01-2.259 1.641l-6.24.907a.2.2 0 00-.111.341l4.515 4.401a3 3 0 01.863 2.655l-1.066 6.215a.2.2 0 00.29.211l5.581-2.934a3.003 3.003 0 012.792 0l5.581 2.934a.2.2 0 00.27-.084c.02-.039.028-.084.02-.127L22.2 19.482a3 3 0 01.863-2.655l4.515-4.401a.2.2 0 00-.11-.341l-6.24-.907a3.001 3.001 0 01-2.259-1.641l-2.791-5.655a.2.2 0 00-.359 0l-2.791 5.655z"
            />
          </svg>
          
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
    <span
      aria-describedby={null}
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiCheckbox-root RcCheckbox-root c1"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
      title="favorite"
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          className="PrivateSwitchBase-input-18"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c2 star_border icon"
        >
          <svg
            className="c3"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 00.753.547l6.641.965a2.001 2.001 0 011.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 01-2.902 2.108l-5.94-3.123a1.002 1.002 0 00-.931 0l-5.94 3.123a2 2 0 01-2.902-2.109l1.134-6.614a.999.999 0 00-.288-.885l-4.806-4.684a2 2 0 011.109-3.411l6.641-.965a1 1 0 00.753-.547l2.97-6.018a2 2 0 012.679-.908zM13.03 9.537a3.001 3.001 0 01-2.259 1.641l-6.24.907a.2.2 0 00-.111.341l4.515 4.401a3 3 0 01.863 2.655l-1.066 6.215a.2.2 0 00.29.211l5.581-2.934a3.003 3.003 0 012.792 0l5.581 2.934a.2.2 0 00.27-.084c.02-.039.028-.084.02-.127L22.2 19.482a3 3 0 01.863-2.655l4.515-4.401a.2.2 0 00-.11-.341l-6.24-.907a3.001 3.001 0 01-2.259-1.641l-2.791-5.655a.2.2 0 00-.359 0l-2.791 5.655z"
            />
          </svg>
          
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiCheckbox-root RcCheckbox-root c1"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
      title="favorite"
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          className="PrivateSwitchBase-input-18"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c2 star_border icon"
        >
          <svg
            className="c3"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 00.753.547l6.641.965a2.001 2.001 0 011.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 01-2.902 2.108l-5.94-3.123a1.002 1.002 0 00-.931 0l-5.94 3.123a2 2 0 01-2.902-2.109l1.134-6.614a.999.999 0 00-.288-.885l-4.806-4.684a2 2 0 011.109-3.411l6.641-.965a1 1 0 00.753-.547l2.97-6.018a2 2 0 012.679-.908zM13.03 9.537a3.001 3.001 0 01-2.259 1.641l-6.24.907a.2.2 0 00-.111.341l4.515 4.401a3 3 0 01.863 2.655l-1.066 6.215a.2.2 0 00.29.211l5.581-2.934a3.003 3.003 0 012.792 0l5.581 2.934a.2.2 0 00.27-.084c.02-.039.028-.084.02-.127L22.2 19.482a3 3 0 01.863-2.655l4.515-4.401a.2.2 0 00-.11-.341l-6.24-.907a3.001 3.001 0 01-2.259-1.641l-2.791-5.655a.2.2 0 00-.359 0l-2.791 5.655z"
            />
          </svg>
          
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiCheckbox-root RcCheckbox-root c1"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
      title="favorite"
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          className="PrivateSwitchBase-input-18"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="c2 star_border icon"
        >
          <svg
            className="c3"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 00.753.547l6.641.965a2.001 2.001 0 011.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 01-2.902 2.108l-5.94-3.123a1.002 1.002 0 00-.931 0l-5.94 3.123a2 2 0 01-2.902-2.109l1.134-6.614a.999.999 0 00-.288-.885l-4.806-4.684a2 2 0 011.109-3.411l6.641-.965a1 1 0 00.753-.547l2.97-6.018a2 2 0 012.679-.908zM13.03 9.537a3.001 3.001 0 01-2.259 1.641l-6.24.907a.2.2 0 00-.111.341l4.515 4.401a3 3 0 01.863 2.655l-1.066 6.215a.2.2 0 00.29.211l5.581-2.934a3.003 3.003 0 012.792 0l5.581 2.934a.2.2 0 00.27-.084c.02-.039.028-.084.02-.127L22.2 19.482a3 3 0 01.863-2.655l4.515-4.401a.2.2 0 00-.11-.341l-6.24-.907a3.001 3.001 0 01-2.259-1.641l-2.791-5.655a.2.2 0 00-.359 0l-2.791 5.655z"
            />
          </svg>
          
        </span>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
  </div>
</div>
`;

exports[`Storyshots ðŸ–¤ deprecated Components/Buttons/ButtonBar Plain Icon Button Bar 1`] = `
.c4 {
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  pointer-events: none;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
  height: 20px;
  color: #666666;
  border-radius: 50%;
  -webkit-transition: background 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  cursor: pointer;
}

.c1,
.c1 .c2 {
  font-size: 20px;
}

.c1:active,
.c1:active .c2 {
  color: #666666;
}

.c1.focus-visible,
.c1[data-focus-visible-added] {
  background-color: transparent;
}

.c1.focus-visible:active,
.c1[data-focus-visible-added]:active,
.c1.focus-visible:active .c2,
.c1[data-focus-visible-added]:active .c2 {
  color: rgba(102,102,102,0.88);
}

.c1.focus-visible:after,
.c1[data-focus-visible-added]:after {
  content: '';
  position: absolute;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
  box-shadow: 0 0 0 1px #066FAC;
  border-radius: 50%;
  pointer-events: none;
  z-index: 1;
}

.c1.RcIconButton-persistBg {
  background-color: rgba(102,102,102,0.12);
}

.c1.RcIconButton-disabled,
.c1.RcIconButton-disabled .c2 {
  color: #B9B9B9;
}

.c1.RcIconButton-invisible {
  opacity: 0;
  visibility: hidden;
  width: 0;
  height: 0;
}

.c1.RcIconButton-outline {
  border: 1px solid #C7C7C7;
}

.c1.RcIconButton-contained {
  color: #fff;
  background-color: #666666;
}

.c1.RcIconButton-contained.focus-visible,
.c1.RcIconButton-contained[data-focus-visible-added] {
  background-color: rgba(102,102,102,0.84);
}

.c1.RcIconButton-contained.focus-visible,
.c1.RcIconButton-contained[data-focus-visible-added],
.c1.RcIconButton-contained.focus-visible .c2,
.c1.RcIconButton-contained[data-focus-visible-added] .c2 {
  color: #fff;
}

.c1.RcIconButton-contained:active {
  background-color: rgba(102,102,102,0.76);
}

.c1.RcIconButton-contained:active,
.c1.RcIconButton-contained:active .c2 {
  color: #fff;
}

.c1 .RcIconButtonTouchRipple-ripplePulsate {
  border-radius: 0;
  -webkit-animation-name: none;
  animation-name: none;
  opacity: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  white-space: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0px;
}

.c0.c0 > *:nth-child(n + 2) {
  margin-left: 0;
}

@media (hover:hover) and (pointer:fine) {
  .c1:hover {
    background-color: transparent;
  }

  .c1:hover,
  .c1:hover .c2 {
    color: rgba(102,102,102,0.88);
  }
}

@media (hover:hover) and (pointer:fine) {
  .c1.RcIconButton-contained:hover {
    background-color: rgba(102,102,102,0.92);
  }

  .c1.RcIconButton-contained:hover,
  .c1.RcIconButton-contained:hover .c2 {
    color: #fff;
  }
}

<div>
  <div
    className="c0 "
    direction="horizontal"
    onClick={[Function]}
    onKeyUp={[Function]}
    size="medium"
  >
    <button
      aria-describedby={null}
      aria-label="like"
      className="MuiButtonBase-root c1 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="like"
      type="button"
    >
      <span
        className="c2 c3 refresh icon"
      >
        <svg
          className="c4"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 3a1 1 0 011 1v6a1 1 0 01-.883.993L27 11h-6a1 1 0 010-2l3.486-.001A10.981 10.981 0 0016 5C9.925 5 5 9.925 5 16s4.925 11 11 11 11-4.925 11-11a1 1 0 012 0c0 7.18-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3a12.98 12.98 0 0110.001 4.693L26 4a1 1 0 011-1z"
          />
        </svg>
        
      </span>
    </button>
    <button
      aria-describedby={null}
      aria-label="like"
      className="MuiButtonBase-root c1 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="like"
      type="button"
    >
      <span
        className="c2 c3 refresh icon"
      >
        <svg
          className="c4"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 3a1 1 0 011 1v6a1 1 0 01-.883.993L27 11h-6a1 1 0 010-2l3.486-.001A10.981 10.981 0 0016 5C9.925 5 5 9.925 5 16s4.925 11 11 11 11-4.925 11-11a1 1 0 012 0c0 7.18-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3a12.98 12.98 0 0110.001 4.693L26 4a1 1 0 011-1z"
          />
        </svg>
        
      </span>
    </button>
    <button
      aria-describedby={null}
      aria-label="like"
      className="MuiButtonBase-root c1 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="like"
      type="button"
    >
      <span
        className="c2 c3 refresh icon"
      >
        <svg
          className="c4"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 3a1 1 0 011 1v6a1 1 0 01-.883.993L27 11h-6a1 1 0 010-2l3.486-.001A10.981 10.981 0 0016 5C9.925 5 5 9.925 5 16s4.925 11 11 11 11-4.925 11-11a1 1 0 012 0c0 7.18-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3a12.98 12.98 0 0110.001 4.693L26 4a1 1 0 011-1z"
          />
        </svg>
        
      </span>
    </button>
    <button
      aria-describedby={null}
      aria-label="like"
      className="MuiButtonBase-root c1 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="like"
      type="button"
    >
      <span
        className="c2 c3 refresh icon"
      >
        <svg
          className="c4"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 3a1 1 0 011 1v6a1 1 0 01-.883.993L27 11h-6a1 1 0 010-2l3.486-.001A10.981 10.981 0 0016 5C9.925 5 5 9.925 5 16s4.925 11 11 11 11-4.925 11-11a1 1 0 012 0c0 7.18-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3a12.98 12.98 0 0110.001 4.693L26 4a1 1 0 011-1z"
          />
        </svg>
        
      </span>
    </button>
  </div>
</div>
`;

exports[`Storyshots ðŸ–¤ deprecated Components/Buttons/ButtonBar Round Icon Button Bar 1`] = `
.c4 {
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  pointer-events: none;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 40px;
  height: 40px;
  color: #666666;
  border-radius: 50%;
  -webkit-transition: background 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  cursor: pointer;
}

.c1,
.c1 .c2 {
  font-size: 20px;
}

.c1:active,
.c1:active .c2 {
  color: #666666;
}

.c1.focus-visible,
.c1[data-focus-visible-added] {
  background-color: rgba(102,102,102,0.16);
}

.c1.focus-visible:active,
.c1[data-focus-visible-added]:active,
.c1.focus-visible:active .c2,
.c1[data-focus-visible-added]:active .c2 {
  color: rgba(102,102,102,0.88);
}

.c1.RcIconButton-persistBg {
  background-color: rgba(102,102,102,0.12);
}

.c1.RcIconButton-disabled,
.c1.RcIconButton-disabled .c2 {
  color: #B9B9B9;
}

.c1.RcIconButton-invisible {
  opacity: 0;
  visibility: hidden;
  width: 0;
  height: 0;
}

.c1.RcIconButton-outline {
  border: 1px solid #C7C7C7;
}

.c1.RcIconButton-contained {
  color: #fff;
  background-color: #666666;
}

.c1.RcIconButton-contained.focus-visible,
.c1.RcIconButton-contained[data-focus-visible-added] {
  background-color: rgba(102,102,102,0.84);
}

.c1.RcIconButton-contained.focus-visible,
.c1.RcIconButton-contained[data-focus-visible-added],
.c1.RcIconButton-contained.focus-visible .c2,
.c1.RcIconButton-contained[data-focus-visible-added] .c2 {
  color: #fff;
}

.c1.RcIconButton-contained:active {
  background-color: rgba(102,102,102,0.76);
}

.c1.RcIconButton-contained:active,
.c1.RcIconButton-contained:active .c2 {
  color: #fff;
}

.c1 .RcIconButtonTouchRipple-ripplePulsate {
  border-radius: 0;
  -webkit-animation-name: none;
  animation-name: none;
  opacity: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  white-space: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0px;
}

.c0.c0 > *:nth-child(n + 2) {
  margin-left: 0;
}

@media (hover:hover) and (pointer:fine) {
  .c1:hover {
    background-color: rgba(102,102,102,0.08);
  }

  .c1:hover,
  .c1:hover .c2 {
    color: rgba(102,102,102,0.88);
  }
}

@media (hover:hover) and (pointer:fine) {
  .c1.RcIconButton-contained:hover {
    background-color: rgba(102,102,102,0.92);
  }

  .c1.RcIconButton-contained:hover,
  .c1.RcIconButton-contained:hover .c2 {
    color: #fff;
  }
}

<div>
  <div
    className="c0 "
    direction="horizontal"
    onClick={[Function]}
    onKeyUp={[Function]}
    size="medium"
  >
    <button
      aria-describedby={null}
      aria-label="like"
      className="MuiButtonBase-root c1 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="like"
      type="button"
    >
      <span
        className="c2 c3 refresh icon"
      >
        <svg
          className="c4"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 3a1 1 0 011 1v6a1 1 0 01-.883.993L27 11h-6a1 1 0 010-2l3.486-.001A10.981 10.981 0 0016 5C9.925 5 5 9.925 5 16s4.925 11 11 11 11-4.925 11-11a1 1 0 012 0c0 7.18-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3a12.98 12.98 0 0110.001 4.693L26 4a1 1 0 011-1z"
          />
        </svg>
        
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-describedby={null}
      aria-label="like"
      className="MuiButtonBase-root c1 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="like"
      type="button"
    >
      <span
        className="c2 c3 refresh icon"
      >
        <svg
          className="c4"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 3a1 1 0 011 1v6a1 1 0 01-.883.993L27 11h-6a1 1 0 010-2l3.486-.001A10.981 10.981 0 0016 5C9.925 5 5 9.925 5 16s4.925 11 11 11 11-4.925 11-11a1 1 0 012 0c0 7.18-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3a12.98 12.98 0 0110.001 4.693L26 4a1 1 0 011-1z"
          />
        </svg>
        
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-describedby={null}
      aria-label="like"
      className="MuiButtonBase-root c1 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="like"
      type="button"
    >
      <span
        className="c2 c3 refresh icon"
      >
        <svg
          className="c4"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 3a1 1 0 011 1v6a1 1 0 01-.883.993L27 11h-6a1 1 0 010-2l3.486-.001A10.981 10.981 0 0016 5C9.925 5 5 9.925 5 16s4.925 11 11 11 11-4.925 11-11a1 1 0 012 0c0 7.18-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3a12.98 12.98 0 0110.001 4.693L26 4a1 1 0 011-1z"
          />
        </svg>
        
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-describedby={null}
      aria-label="like"
      className="MuiButtonBase-root c1 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="like"
      type="button"
    >
      <span
        className="c2 c3 refresh icon"
      >
        <svg
          className="c4"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27 3a1 1 0 011 1v6a1 1 0 01-.883.993L27 11h-6a1 1 0 010-2l3.486-.001A10.981 10.981 0 0016 5C9.925 5 5 9.925 5 16s4.925 11 11 11 11-4.925 11-11a1 1 0 012 0c0 7.18-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3a12.98 12.98 0 0110.001 4.693L26 4a1 1 0 011-1z"
          />
        </svg>
        
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
